### ISSUE TITLE
Memory leaking in version 3.1.8

### ISSUE THREAD

**Issue by simonmcl** [airbnb/lottie-ios#1234]
##  This issue is a:  
- [x] Crashing Bug

## Which Version of Lottie are you using?

Lottie 3.1.8

## What Platform are you on? 
- [x] iOS

## What Language are you in?
- [x] Swift

## Expected Behavior 
Not to leak memory

## Actual Behavior

Every viewController that has a lottie animation is leaking memory and no others. Many only load a lottie image and nothing else, acting as a success screen to let the user know an action is done.

I've tried clearing the cache on viewDidDisappear and it didn't help. The memory leaks without transitioning to another screen. Seems to leak as soon as the animation starts playing. Googling the error, I see similar inspector logs, with the problem being how the files are being fetched. One such example suggests to not use `init(withContentsOfFile: )`: https://stackoverflow.com/questions/61832483/uiimageview-instances-of-nspathstore2-leaked

![Screenshot 2020-07-24 at 10 47 01](https://user-images.githubusercontent.com/8058386/88379518-070ae180-cd9b-11ea-9aa6-35d03159a2ba.png)

## Code Example
Setting the Animation view via storyboard and then calling this code:

```
import UIKit
import Lottie

class OnboardingSuccessViewController: UIViewController {
	
	@IBOutlet weak var animationView: AnimationView?
	
	override func viewDidLoad() {
		super.viewDidLoad()
		
		if let aView = animationView {
			let animation = Animation.named("lottie-success")
			aView.animation = animation
		}
	}
	
	override func viewDidAppear(_ animated: Bool) {
		super.viewDidAppear(animated)
		
		if let aView = animationView {
			aView.play()
		}
	}
}
```

## Animation JSON

Include the use of 5 images, these are placed in the folder next to the animation json file:

```
{"v":"5.6.5","fr":30,"ip":0,"op":60,"w":216,"h":300,"nm":"Comp 1","ddd":0,"assets":[{"id":"image_0","w":312,"h":312,"u":"images/","p":"img_0.png","e":0},{"id":"image_1","w":792,"h":973,"u":"images/","p":"img_1.png","e":0},{"id":"image_2","w":1584,"h":893,"u":"images/","p":"img_2.png","e":0},{"id":"image_3","w":1584,"h":482,"u":"images/","p":"img_3.png","e":0},{"id":"image_4","w":1584,"h":1656,"u":"images/","p":"img_4.png","e":0}],"layers":[{"ddd":0,"ind":1,"ty":2,"nm":"Success landing","refId":"image_0","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":1,"k":[{"i":{"x":[0.667],"y":[1]},"o":{"x":[0.167],"y":[0.167]},"t":54,"s":[-90]},{"t":59,"s":[0]}],"ix":10},"p":{"a":1,"k":[{"i":{"x":0.667,"y":1},"o":{"x":0.167,"y":0.167},"t":54,"s":[108,-102,0],"to":[0,51.667,0],"ti":[0,-51.667,0]},{"t":59,"s":[108,208,0]}],"ix":2},"a":{"a":0,"k":[156,156,0],"ix":1},"s":{"a":0,"k":[57,57,100],"ix":6}},"ao":0,"ip":45,"op":60,"st":18,"bm":0},{"ddd":0,"ind":2,"ty":2,"nm":"orange.png","cl":"png","refId":"image_1","sr":1,"ks":{"o":{"a":1,"k":[{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":8,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":13,"s":[100]},{"t":37,"s":[0]}],"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[72.25,179.125,0],"ix":2},"a":{"a":0,"k":[396,486.5,0],"ix":1},"s":{"a":0,"k":[9,9,100],"ix":6}},"ao":0,"ip":0,"op":60,"st":0,"bm":0},{"ddd":0,"ind":3,"ty":2,"nm":"red.png","cl":"png","refId":"image_2","sr":1,"ks":{"o":{"a":1,"k":[{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":5,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":10,"s":[100]},{"t":31,"s":[0]}],"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[108.25,182.75,0],"ix":2},"a":{"a":0,"k":[792,446.5,0],"ix":1},"s":{"a":0,"k":[9,9,100],"ix":6}},"ao":0,"ip":0,"op":60,"st":0,"bm":0},{"ddd":0,"ind":4,"ty":2,"nm":"purple.png","cl":"png","refId":"image_3","sr":1,"ks":{"o":{"a":1,"k":[{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":3,"s":[0]},{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":7,"s":[100]},{"t":27,"s":[0]}],"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[108,201.25,0],"ix":2},"a":{"a":0,"k":[792,241,0],"ix":1},"s":{"a":0,"k":[9,9,100],"ix":6}},"ao":0,"ip":0,"op":60,"st":0,"bm":0},{"ddd":0,"ind":5,"ty":2,"nm":"icon.png","cl":"png","refId":"image_4","sr":1,"ks":{"o":{"a":0,"k":100,"ix":11},"r":{"a":0,"k":0,"ix":10},"p":{"a":0,"k":[108,210,0],"ix":2},"a":{"a":0,"k":[792,828,0],"ix":1},"s":{"a":0,"k":[9,9,100],"ix":6}},"ao":0,"ip":0,"op":60,"st":0,"bm":0},{"ddd":0,"ind":6,"ty":2,"nm":"Success erupt","refId":"image_0","sr":1,"ks":{"o":{"a":1,"k":[{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.167],"y":[0.167]},"t":16,"s":[0]},{"t":23,"s":[100]}],"ix":11},"r":{"a":1,"k":[{"i":{"x":[0.833],"y":[0.833]},"o":{"x":[0.333],"y":[0]},"t":14,"s":[0]},{"t":24,"s":[180]}],"ix":10},"p":{"a":1,"k":[{"i":{"x":0.833,"y":0.833},"o":{"x":0.333,"y":0},"t":14,"s":[111,178,0],"to":[0,-35,0],"ti":[0,35,0]},{"t":24,"s":[111,-32,0]}],"ix":2},"a":{"a":0,"k":[156,156,0],"ix":1},"s":{"a":1,"k":[{"i":{"x":[0.833,0.833,0.833],"y":[0.833,0.833,0.833]},"o":{"x":[0.167,0.167,0.167],"y":[0.167,0.167,0.167]},"t":14,"s":[5,5,100]},{"t":24,"s":[13,13,100]}],"ix":6}},"ao":0,"ip":14,"op":25,"st":-33,"bm":0}],"markers":[]}
```

**Comment by vapor-pawelw** [comment]
I just found that `textProvider` and `imageProvider` in `AnimationView` are acting as delegates and their protocols require them to be classes but they are not marked as `weak`. Once I set it `textProvider` to `DefaultTextProvider()` just before dismissing my controller using lottie, my own view wrapping animationView deallocated properly. This is a serious issue and should be resolved as quickly as possible.
